credentials/apps/verifiable_credentials/permissions.py

```diff
@@ -22,21 +32,34 @@ def has_permission(self, request, view):
             "type": [
                 "VerifiablePresentation"
             ],
-            "holder": "<issuer-did>",
+            "holder": "<holder-did>",  <-- generated by a Wallet subject/holder ID
             "proof": {
                 "type": "Ed25519Signature2020",
                 "created": "2023-04-03T13:18:05Z",
-                "verificationMethod": "<issuer-did>",
+                "verificationMethod": "<issuer-did>",  <-- Issuer ID from deeplink params
                 "proofPurpose": "authentication",
-                "challenge": "c44c45c6-e6e1-4db1-ac2d-413aa0eaf438",
+                "challenge": "c44c45c6-e6e1-4db1-ac2d-413aa0eaf438",  <-- IssuanceLine.uuid from deeplink params
                 "proofValue": "z5B2bRwmvzoAfJ8JnsnnXCdAMBk28nW9NY8eVm83HSq92MkA8rCq2SVx58pwYiTzZFQNb2hHt3NEc1DQ..."
             }
         }
         """
+        presentation = request.data
+
+        # validate proof:
+        proof = presentation.get("proof", {})
+        if not proof.get("proofPurpose") == "authentication":
+            return False
+        if not IssuanceLine.objects.filter(uuid=proof.get("challenge")).exists():
+            return False
 
+        # verify presentation:
         try:
-            if "VerifiablePresentation" in request.data["type"]:
-                return True
-        except (KeyError, TypeError, IndexError):
+            presentation_json = json.dumps(presentation)
+            proof_options_json = json.dumps({})
+            result = didkit_verify_presentation(presentation_json, proof_options_json)
+            logger.debug("Verifiable presentiation authN result: (%s)", result)
+            return True
+        except didkit.DIDKitException:  # pylint: disable=no-member
             pass
```

Comment:
Maybe let's include log exception already, as this code is close to the final version?